lib/docker-toolbelt.coffee

```diff
@@ -42,8 +49,23 @@ createChainIdFromParent = (parent, dgsts) ->
 
 	return createChainIdFromParent(dgst, dgsts[1..])
 
+getDiffIds = (dkroot, driver, imageId) ->
+	[ hashType, hash ] = imageId.split(':')
+	fs.readFileAsync(path.join(dkroot, "image/#{driver}/imagedb/content", hashType, hash))
+	.then(JSON.parse)
+	.get('rootfs').get('diff_ids')
+
+getCacheId = (dkroot, driver, layerId) ->
+	[ hashType, hash ] = layerId.split(':')
```

Comment:
This could throw if `layerId` is not a string (which would be a bit of a surprise since it normally returns a promise)